### ISSUE TITLE
Feed validation recommendations via W3

### ISSUE THREAD

**Issue by infopetal** [diffbot/rss-anything#2]
I am trying to add an RSS feed created with RSS Anything to [Slack](https://slack.com/help/articles/218688467-Add-RSS-feeds-to-Slack), but I keep receiving the error "Error fetching RSS feed." Their [help page](https://slack.com/help/articles/218688467-Add-RSS-feeds-to-Slack#:~:text=Note%3A%20If%20you%20get%20an%20error%20when%20trying%20to%20add%20a%20feed%2C%20try%20validating%20the%20feed%20at%20http%3A//validator.w3.org/feed/.) suggests to try validating the feed at [http://validator.w3.org/feed/](validator.w3.org/feed), and I tried two different RSS Anything–generated feeds from two different sites and received the same error:

```
This feed is valid, but interoperability with the widest range of feed readers could be improved by implementing the following recommendations.

line 2, column 446: Email address is missing real name [[help](https://validator.w3.org/feed/docs/warning/MissingRealName.html)]

... dDate><managingEditor>jerome@diffbot.com</managingEditor><item><title>Cl ...
                                             ^
line 32, column 1494: Missing atom:link with rel="self" [[help](https://validator.w3.org/feed/docs/warning/MissingAtomSelfLink.html)]

... 87f-497a-aabd-14947547d850</guid></item></channel></rss>
```

Are these recommendations possible to implement? Any other insights would be welcome!

With gratitude,
Karly

**Comment by jeromechoo** [comment]
Hey Karly, thanks for submitting this issue. I've made the changes requested from W3C and [verified that it passes W3C](https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Frss.diffbot.com%2Frss%3Furl%3Dhttps%3A%2F%2Fwww.theatlantic.com%2Fcategory%2Fwork-progress%2F) with this commit: https://github.com/diffbot/rss-anything/commit/7702579a6cc361ecc02b29cad35fb80d9a64f863

That said, I'm doubtful that these details are what is causing the RSS feed on Slack to fail. I can't test the RSS integration on my Slack, but I did find [this other similar Github issue](https://github.com/GoogleChrome/developer.chrome.com/issues/2680#issuecomment-1122967034) that cites these potential reasons:

 - There must be a supported data tag.
 - Items cannot be listed out of sequence.
 - Duplicate dates cannot exist.

I suspect that it may be due to duplicate dates. I'm not sure what page you're trying to RSSify, but if two posts in the feed has the same date (without a timestamp), it might be the cause.

RSS Anything does support full timestamps, but only if the source also includes it. If it doesn't, there's not much I can do.

**Comment by infopetal** [comment]
Thank you for the quick response, @jeromechoo! I tried generating and adding my feed to Slack again, and it worked this time, despite duplicate dates existing. I'll be curious to see if/how well it posts to Slack knowing the limitations you found on the other GitHub issue. Whether or not the feed works in Slack in the long run, I've been able to follow RSS Anything feeds in my RSS reader successfully for several weeks and am very grateful for your creating and maintaining it.
